build.gradle

```diff
@@ -50,21 +50,37 @@ subprojects {
       javadoc = true
       tests = false
     }
+    nexus {
+      def gitProvider = new nebula.plugin.info.scm.GitScmProvider()
+      def shortHash = gitProvider.calculateChange(new File('.'))
+      def isRelease = System.getProperty('build.snapshot', 'true') == 'false'
+      if (isRelease) {
+        repositoryUrl = System.getProperty('build.repository', "file://${System.getenv('HOME')}/elasticsearch-releases/${version}-${shortHash}/")
+      }
+    }
     // we have our own username/password prompts so that they only happen once
-    // TODO: add gpg signing prompts
+    // TODO: add gpg signing prompts, which is tricky, as the buildDeb/buildRpm tasks are executed before this code block
```

Comment:
sorry for the confusion.. what I meant is that the configuration for the packages in `ospackage` is executed before this `whenReady` code block and thus reading a key here and applying it is tricky. I guess we can do some in-place replace magic?